docs/advanced.md

```diff
@@ -403,6 +403,51 @@ some of yargs' parsing features:
   }
 }
 ```
+## Midlewares
+Sometimes you want to perform operations on the args just before reaching the command.
+For example, you want to validate the args or check if the an argument is passed otherwise you load it from local file.
+Middlwares are basically functions ran before the command to prepare the args for them.
+
+Diagram:
+
+```
+                        --------------         --------------        ---------
+stdin ----> argv ----> | Middleware 1 | ----> | Middleware 2 | ---> | Command |
+                        --------------         --------------        ---------
+```
+### Usage
+
+Here the middleware will check if the `username` and `password` is passed by the user
+if not it will load them from `~./credential` file and fil in the argv.username and argv.password.
```

Comment:
slight grammar nits:

```
In this example, our middleware will check if the `username` and `password` is provided. If not, it will load them from `~/.credentials`, and fill in the `argv.username` and `argv.password` values.
```